components/org.wso2.carbon.identity.oauth.common/src/main/java/org/wso2/carbon/identity/oauth/common/IDTokenResponseValidator.java

```diff
@@ -61,6 +65,7 @@ public void validateRequiredParameters(HttpServletRequest request) throws OAuthP
         // for id_token response type, the scope parameter should contain 'openid' as one of the scopes.
         String openIdScope = request.getParameter(SCOPE);
         if (StringUtils.isBlank(openIdScope) || !containOIDCScope(openIdScope)) {
+            diagnosticLog.error("'response_type' contains 'id_token', but 'openid' scope not found.");
```

Comment:
"The 'response_type' parameter contains 'id_token', but 'openid' scope not found in the request."